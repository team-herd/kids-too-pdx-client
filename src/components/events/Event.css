.Event {
  h2{
    margin: 0;
    font-size: 2.5em;
    text-align: center;
  }
  p {
    text-align: center;
    margin: .4em;
    span {
      font-weight: 800;
      margin-right: .5em;
    }
  }
  img {
    display: block;
    height: 40vw;
    width: auto;
    margin: auto;
    border-radius: 5px;
    box-shadow: 2px 2px gray;;
  }
  
}