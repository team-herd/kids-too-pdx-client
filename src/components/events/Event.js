import React from 'react';
import PropTypes from 'prop-types';
import styles from './Event.css';
import { Link } from 'react-router-dom';

// import dollar from '../../../public/assets/dollar-symbol.png';

function Event({ event }) {
  const { name, date, time, location, cost, age, description, category, picture } = event;
  // if(event.cost === 'free') {
  //   return <img src={dollar} alt="dollar sign"/>;
  // }
  return (

    <>
      <section className={styles.Event}>
        <img src={picture} alt='event picture' />
        <Link to={`/events/${event.id}`} key={event.id}><h2>{name}</h2></Link>
        <p>{date} @ {time}</p>
        <p><span>Location:</span>{location.address}</p>
        <p><span>Cost:</span>{cost}</p>
        <p><span>Age:</span>{age}</p>
        <p>{category}</p>
        <p>{description}</p>
      </section>
    </>
  );
}
Event.propTypes = {
  event: PropTypes.object.isRequired
};
export default Event;
