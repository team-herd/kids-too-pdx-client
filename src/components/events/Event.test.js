import React from 'react';
import renderer from 'react-test-renderer';
import Event from './Event';
import { MemoryRouter } from 'react-router-dom';

describe('Event', () => {
  it('matches snapshot for Event', () => {
    const props = {
      event: {
        _id: 1,
        image: 'image.com',
        time: '2pm',
        name: 'House Event',
        date: new Date().toString(),
        location: {
          venue: 'My House'
        },
        price: 250,
        ageMin: 9,
        ageMax: 12,
        category: 'Art',
        description: 'The bestest event'
      }
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Event {...props} />
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
