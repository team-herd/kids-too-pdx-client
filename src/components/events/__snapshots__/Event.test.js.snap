// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event matches snapshot for Event 1`] = `
<section
  className="Event"
>
  <img
    alt="event picture"
    src="image.com"
  />
  <a
    href="/events/1"
    onClick={[Function]}
  >
    <h2>
      House Event
    </h2>
  </a>
  <p>
    <span>
      Date:
    </span>
    Fri Mar 29 2019 08:53:35 GMT-0700 (Pacific Daylight Time)
  </p>
  <p>
    <span>
      Time:
    </span>
     
    2pm
  </p>
  <p>
    <span>
      Location:
    </span>
    My House
  </p>
  <p>
    <span>
      Cost:
    </span>
    250
  </p>
  <p>
    <span>
      Age:
    </span>
     
    9-12
  </p>
  <p>
    Art
  </p>
</section>
`;
